scripts/run_tests.py

```diff
@@ -7,6 +9,8 @@
 # script.
 # (recipe here)
 # http://stackoverflow.com/questions/4789837/how-to-terminate-a-python-subprocess-launched-with-shell-true
+#
+# ruff: noqa: T201
```

Comment:
when I first started checking through the changes, I made some minor fixes and turned off various linting rules (here, warnings about print statements). I soon realised how much work that would be so that stopped pretty quickly!